### ISSUE TITLE
Race condition in database creation

### ISSUE THREAD

**Issue by matko** [terminusdb/terminusdb#1369]
When creating databases rapidly with the same name, a race condition occurs where the system database label is read incompletely.

It seems like sometimes the system database label file is read containing only one line. This implies that some sort of flush happens between the two lines. They should be written atomically (which should be possible as the whole write is less than a page).

A short term solution here is to just re-enable read locks on labels. But we should investigate why the flush happens at all.

**Comment by matko** [comment]
The following script triggers the behavior:
```
for X in {1..100}; do
    curl -u admin:root "http://localhost:6363/api/db/admin/db${RANDOM}" -d '{ "label": "db", "comment": "random db" }' -H 'Content-Type: application/json' &
done
```

This results in
```
{
  "api:message":"Error: rust_io_error(\"expected label file to have two lines. contents were ([])\")\n  [50] terminus_store:open_named_graph(<store>,'terminusdb%3a%2f%2f%2fsystem%2fdata',_7610)\n  [48] triplestore:safe_open_graph_head_(<store>,'terminusdb:///system/data',_7650) at /app/terminusdb/src/core/triple/triplestore.pl:226\n  [46] descriptor:open_read_write_obj(system_graph{type:instance},read_write_obj{descriptor:system_graph{type:instance},read:_7706,triple_update:false,write:_7710},[],[... = ...]) at /app/terminusdb/src/core/transaction/descriptor.pl:219\n  [44] utils:mapm_([read_write_obj{descriptor: ...,read:_7794,triple_update:false,write:_7798}],[],_7780,database:reset_read_write_obj) at /app/terminusdb/src/core/util/utils.pl:510\n  [42] database:reset_transaction_object_graph_descriptors(transaction_object{descriptor:system_descriptor{},inference_objects:[],instance_objects:[...],schema_objects:[...]},[],_7854) at /app/terminusdb/src/core/transaction/database.pl:156\n  [41] utils:mapm_([transaction_object{descriptor: ...,inference_objects:[],instance_objects: ...,schema_objects: ...}],[],_7930,database:reset_transaction_object_graph_descriptors) at /app/terminusdb/src/core/util/utils.pl:510\n  [37] database:retry_transaction(query_context{all_witnesses:false,authorization:'terminusdb://system/data/User/admin',bindings:[],default_collection:system_descriptor{},files:[],filter:type_filter{types: ...},prefixes:_8062{'@base':\"terminusdb://system/data/\",'@schema':\"http://terminusdb.com/schema/system#\",'@type':'Context',api:'http://terminusdb.com/schema/api#',json:'http://terminusdb.com/schema/json#',owl:'http://www.w3.org/2002/07/owl#',rdf:'http://www.w3.org/1999/02/22-rdf-syntax-ns#',rdfs:'http://www.w3.org/2000/01/rdf-schema#',sys:'http://terminusdb.com/schema/sys#',vio:'http://terminusdb.com/schema/vio#',woql:'http://terminusdb.com/schema/woql#',xdd:'http://terminusdb.com/schema/xdd#',xsd:\"http://www.w3.org/2001/XMLSchema#\"},selected:[],system:system_descriptor{},transaction_objects:[...],update_guard:_8048,write_graph:system_graph{type:instance}},4) at /app/terminusdb/src/core/transaction/database.pl:200\n  [36] database:with_transaction_(query_context{all_witnesses:false,authorization:'terminusdb://system/data/User/admin',bindings:[],default_collection:system_descriptor{},files:[],filter:type_filter{types: ...},prefixes:_8062{'@base':\"terminusdb://system/data/\",'@schema':\"http://terminusdb.com/schema/system#\",'@type':'Context',api:'http://terminusdb.com/schema/api#',json:'http://terminusdb.com/schema/json#',owl:'http://www.w3.org/2002/07/owl#',rdf:'http://www.w3.org/1999/02/22-rdf-syntax-ns#',rdfs:'http://www.w3.org/2000/01/rdf-schema#',sys:'http://terminusdb.com/schema/sys#',vio:'http://terminusdb.com/schema/vio#',woql:'http://terminusdb.com/schema/woql#',xdd:'http://terminusdb.com/schema/xdd#',xsd:\"http://www.w3.org/2001/XMLSchema#\"},selected:[],system:system_descriptor{},transaction_objects:[...],update_guard:_8226,write_graph:system_graph{type:instance}},db_create:(...;...),_8180) at /app/terminusdb/src/core/transaction/database.pl:233\n  [35] setup_call_catcher_cleanup(database:pre_transaction_tabling,database:with_transaction_(...,...,_8386),_8364,database:post_transaction_tabling) at /usr/lib/swipl/boot/init.pl:663\n  [30] '<meta-call>'('<garbage_collected>') <foreign>\n  [29] catch(routes:(...,...),error(rust_io_error(\"expected label file to have two lines. contents were ([])\"),context(_8486,_8488)),routes:do_or_die(...,...)) at /usr/lib/swipl/boot/init.pl:562\n  [28] catch_with_backtrace('<garbage_collected>','<garbage_collected>','<garbage_collected>') at /usr/lib/swipl/boot/init.pl:629\n\nNote: some frames are missing due to last-call optimization.\nRe-run your program in debug mode (:- debug.) to get more detail.\n\n",
  "api:status":"api:server_error"
}
```

**Comment by rrooij** [comment]
Seems like the problem is not reproducible anymore after enabling the read locks in the reenable_read_locks branch.

**Comment by rrooij** [comment]
Fixed in: a2d1f908a0f5718ca4850d96bc94297b7ace2c32
